bitsail-connectors/connector-redis/src/main/java/com/bytedance/bitsail/connector/redis/sink/RedisWriter.java

```diff
@@ -0,0 +1,249 @@
+    columnSize = redisOptions.getColumnSize();
+    complexTypeWithTtl = redisOptions.isComplexTypeWithTtl();
+    commandDescription = redisOptions.getCommandDescription();
+    logSampleInterval = redisOptions.getLogSampleInterval();
+
+    JedisPoolConfig jedisPoolConfig = new JedisPoolConfig();
+    jedisPoolConfig.setMaxTotal(RedisConstants.DEFAULT_MAX_TOTAL_CONNECTIONS);
+    jedisPoolConfig.setMaxIdle(RedisConstants.DEFAULT_MAX_IDLE_CONNECTIONS);
+    jedisPoolConfig.setMinIdle(RedisConstants.DEFAULT_MIN_IDLE_CONNECTIONS);
+    jedisPoolConfig.setMaxWaitMillis(RedisConstants.DEFAULT_MAX_WAIT_TIME_IN_MILLS);
+
+    String redisPassword = redisOptions.getRedisPassword();
+    String redisHost = redisOptions.getRedisHost();
+    int redisPort = redisOptions.getRedisPort();
+    int timeout = redisOptions.getTimeout();
+    if (StringUtils.isEmpty(redisPassword)) {
+      this.jedisPool = new JedisPool(jedisPoolConfig, redisHost, redisPort, timeout);
+    } else {
+      this.jedisPool = new JedisPool(jedisPoolConfig, redisHost, redisPort, timeout, redisPassword);
+    }
+
+    this.jedisFetcher = RetryerBuilder.<Jedis>newBuilder()
+        .retryIfResult(Objects::isNull)
+        .retryIfRuntimeException()
+        .withStopStrategy(StopStrategies.stopAfterAttempt(3))
+        .withWaitStrategy(WaitStrategies.exponentialWait(100, 5, TimeUnit.MINUTES))
+        .build()
+        .wrap(jedisPool::getResource);
+    this.retryer = RetryerBuilder.<Boolean>newBuilder()
+        .retryIfResult(needRetry -> Objects.equals(needRetry, true))
+        .retryIfException(e -> !(e instanceof BitSailException))
+        .withWaitStrategy(WaitStrategies.fixedWait(3, TimeUnit.SECONDS))
+        .withStopStrategy(StopStrategies.stopAfterAttempt(MAX_ATTEMPT_NUM))
+        .build();
+  }
+
+  @Override
+  public void write(Row record) throws IOException {
+    validate(record);
+    this.recordQueue.add(record);
+    if (recordQueue.isAtFullCapacity()) {
+      flush(false);
+    }
+  }
+
+  /**
+   * Pre-check data.
+   */
+  protected void validate(Row record) throws BitSailException {
+    if (record.getArity() != columnSize) {
+      throw BitSailException.asBitSailException(CommonErrorCode.CONFIG_ERROR,
+          String.format("The record's size is %d , but supposed to be %d!", record.getArity(), columnSize));
+    }
+    for (int i = 0; i < columnSize; i++) {
+      if (record.getField(i) == null) {
+        throw BitSailException.asBitSailException(CommonErrorCode.UNSUPPORTED_ENCODING,
+            String.format("record contains null element in index[%d]", i));
+      }
+    }
+    if (commandDescription.getJedisCommand() == JedisCommand.ZADD) {
+      parseScoreFromBytes((byte[]) record.getField(1));
+    }
+  }
+
+  /**
+   * check if score field can be parsed to double.
+   */
+  @VisibleForTesting
+  public double parseScoreFromBytes(byte[] scoreInBytes) throws BitSailException {
```

Comment:
please add a UT for this method

Reply:
I delete this @VisibleForTesting tag and move the method to private, because what the method do is too easy and clearly so I don't migrate the test method, it just do Double.parseDouble(new String(scoreInBytes)).